frontend/src/lib/requests.ts

```diff
@@ -0,0 +1,34 @@
+import { API_BASE_URL, log } from '.';
+
+import type { AxiosRequestConfig } from 'axios';
+import axios from 'axios';
+
+const quizootApi = axios.create({
+    baseURL: API_BASE_URL,
+    timeout: 1000,
+    maxBodyLength: 2000,
+});
+
+export type ApiResponse =
+    | number
+    | string
+    | boolean
+    | null
+    | ApiResponse[]
+    | { [x: string]: ApiResponse };
+
+export async function makeRequest<TData extends ApiResponse>(
+    config: AxiosRequestConfig
+): Promise<TData> {
+    try {
+        const response = await quizootApi(config);
+        return response.data;
+    } catch (err) {
+        if (axios.isAxiosError(err)) {
+            // TODO: display
```

Comment:
For axios error, we can probably do something better with the error message

Reply:
I don't know for now. "Maybe" you can find any idea about that in this [article](https://dev.to/mperon/axios-error-handling-like-a-boss-333d)